src/ir_passes/mono/monodeferred.rs

```diff
@@ -266,6 +266,26 @@ impl<'a, 'pass: 'a> MonoDeferred<'a, 'pass> {
         self.monosig.base.cmds.extend(body);
     }
 
+    fn p_let(&mut self, l: &ir::Let) {
+        let ir::Let { param, expr } = *l;
+
+        let expr = self.monosig.expr(
+            self.underlying,
+            self.pass,
+            Underlying::new(expr),
+        );
+
+        // Inserts this param into the binding.
+        self.monosig.binding.insert(
+            Underlying::new(param),
+            self.monosig
+                .base
+                .bin(self.monosig.base.get(expr.idx()).clone())
+                .as_concrete(&self.monosig.base)
+                .unwrap(),
+        );
+    }
```

Comment:
Not sure if this works when you have a local parameter dependent on a loop index.